integration/ctr/oci_hooks_tests.bats

```diff
@@ -0,0 +1,78 @@
+#!/usr/bin/env bats
+#
+# Copyright (c) 2022 Apple Inc.
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+
+IMAGE="${IMAGE:-docker.io/library/busybox:latest}"
+CONTAINER_NAME="${CONTAINER_NAME:-kataocihookstest}"
+CTR_RUNTIME="${CTR_RUNTIME:-io.containerd.kata.v2}"
+
+set -o errexit
+set -o nounset
+set -o pipefail
+set -o errtrace
+
+readonly KATA_CONFIG="${KATA_CONFIG:-/etc/kata-containers/configuration.toml}"
+readonly KATA_CONFIG_BACKUP="$KATA_CONFIG.backup"
+readonly DEFAULT_KATA_CONFIG="/usr/share/defaults/kata-containers/configuration.toml"
+
+function teardown() {
+	sudo rm -rf $HOST_TMP_TEST_DIR
+	[ -f "$KATA_CONFIG_BACKUP" ] && sudo mv "$KATA_CONFIG_BACKUP" "$KATA_CONFIG" || \
+		sudo cp "$DEFAULT_KATA_CONFIG" "$KATA_CONFIG"
+}
+
+function setup() {
+	HOST_TMP_TEST_DIR="$(mktemp -d)"
+	sudo mkdir -p $(dirname "${KATA_CONFIG}")
+	[ -f "$KATA_CONFIG" ] && sudo cp "$KATA_CONFIG" "$KATA_CONFIG_BACKUP" || \
```

Comment:
Let `KATA_CONFIG=/etc/kata-containers/configuration.toml` and `$KATA_CONFIG` file doesn't exist. Then:
1. `setup` copies `/usr/share/defaults/kata-containers/configuration.toml` to `$KATA_CONFIG`
2. This test changes `$KATA_CONFIG`
3. `teardown` copies `/usr/share/defaults/kata-containers/configuration.toml` to `$KATA_CONFIG`
4. This test exited and left a `/etc/kata-containers/configuration.toml` behind.
5. The next test doesn't change the `/etc/kata-containers/configuration.toml` because this file is unexpected on CI by default. Instead it looks at `/usr/share/defaults/kata-containers/configuration.toml` but changes take no effect anymore.

It seems error prone. Am I wrong and/or too paranoiac?

Maybe we could turn this method https://github.com/kata-containers/tests/blob/main/functional/tdx/run.sh#L39  generic then on this test you get the default config, save it on setup, change on test, and restore on teardown.

Reply:
I think you may have a point -- I do want to mention I used the same pattern as [here](https://github.com/kata-containers/tests/blob/2.4.1/integration/containerd/cri/integration-tests.sh#L65-L66), and I trusted the assumptions made there. I can't see the function you're linking to, since `main` has changed!